libraries/AP_HAL_ChibiOS/UARTDriver.cpp

```diff
@@ -763,6 +763,10 @@ ssize_t UARTDriver::_read(uint8_t *buffer, uint16_t count)
         update_rts_line();
     }
 
+#if HAL_UART_DEBUG_LOGGING_ENABLED
+    log_read_data(buffer, count);
+#endif
```

Comment:
Have you noticed any problem for our drivers which read 1 byte at a time? That will result in a lot of overhead. Perhaps the logging thread could do some sort of linearization?

Reply:
To start with we would coalesce within the same millisecond and direction, that wouldn't create a meaningful change. But it could be deferred.